### ISSUE TITLE
New Connection ID frame repeated with different content

### ISSUE THREAD

**Issue by huitema** [private-octopus/picoquic#1319]
I am looking at the interop runner failed test C1 between neqo client and picoquic server. One of the connections failed, see [client log](https://interop.seemann.io/logs/2022-03-15T08:08/picoquic_neqo/handshakecorruption/client/qlog/server4-24bfa124caa8180a.qlog) and [server log](https://interop.seemann.io/logs/2022-03-15T08:08/picoquic_neqo/handshakecorruption/server/qlog/24bfa124caa8180a.server.qlog). What seems to happen is:

* Picoquic server sends 7 new connection ID frames in 1 RTT packet number 0, with CID number 1 to 7
* Packet number 0 is marked lost after some time
* Content of packet #0 is repeated in packet number 3
* Packet number 3 is acked by client.
* Picoquic repeats New CID number 1 to 7 in packet number 6 but with different values.
* Neqo client detects the issue and closes the connection.

**Comment by huitema** [comment]
Same issue reproes in a test between quic-go client and pricoquic server. See [client log](https://interop.seemann.io/logs/2022-03-23T08:08/picoquic_quic-go/handshakeloss/client/qlog/430292ad14cb182e7fd82d63.qlog) and [server log](https://interop.seemann.io/logs/2022-03-23T08:08/picoquic_quic-go/handshakeloss/client/qlog/430292ad14cb182e7fd82d63.qlog).

**Comment by huitema** [comment]
Issue is hard to repro, points to some random behavior.

The creation of local connection ID is done in `picoquic_create_local_cnxid`. The current code manages two counters: `cnx->local_cnxid_sequence_next` and `cnx->nb_local_cnxid`.

The `new_connection_id` frames are added to outgoing packets in the function `picoquic_format_new_local_id_as_needed`. The code uses `cnx->nb_local_cnxid` to check whether new local CID should be created:
```
        cnx->nb_local_cnxid < ((int)(cnx->remote_parameters.active_connection_id_limit) + cnx->nb_local_cnxid_expired) &&
        cnx->nb_local_cnxid <= (PICOQUIC_NB_PATH_TARGET+cnx->nb_local_cnxid_expired)
```
This code assumes that `cnx->nb_local_cid` describes the total number of connection ID sent to the peer since the beginning of the connection. If the test passes, the connection ID is created by calling `picoquic_create_local_cnxid`. If the function fails, it has no side effect. If is succeeds, it creates a new CID, with the CID sequence number set as:
```
            l_cid->sequence = cnx->local_cnxid_sequence_next++;
            cnx->nb_local_cnxid++;
```
There is a potential error condition if the code creates a CID, and then finds that there is not enough space in the packet to send it. In that case, the creation is reversed by doing:
```
                /* Oops. Try again next time. */
                picoquic_delete_local_cnxid(cnx, l_cid);
                cnx->local_cnxid_sequence_next--; 
```
When a local CID is deleted in `picoquic_delete_local_cnxid`, the counter is decremented:
```
            cnx->nb_local_cnxid--;
```
This analysis points to two issues:

1) The variables `nb_local_cnxid` and `local_cnxid_sequence_next` are redundant. They may become out of synch, which is not too good.
2) The "number of CID created since the beginning of the connection" should never be decremented, unless when rolling back an immediate creation.

**Comment by huitema** [comment]
Crash observed by @victorstewart in:

![image](https://user-images.githubusercontent.com/7464128/160252791-fbf9f89f-473e-45b1-8066-4e896ef94869.png)

Repro: 
startup the server, client attempts to connect, we receive the 1200 byte client initial then respond with the 1232 byte packet then 110 byte packet…. then we timeout (no further client packers yet)… so my event loop breaks out and i call picoquic_prepare_next_packet

Packet size TP is 1364 on both sides.

**Comment by huitema** [comment]
Probably fixed in PR #1335
